syft/cataloger/java/archive_parser.go

```diff
@@ -178,36 +178,40 @@ func (j *archiveParser) discoverPkgsFromPomProperties(parentPkg *pkg.Package) ([
 			return nil, fmt.Errorf("failed to parse pom.properties (%s): %w", j.virtualPath, err)
 		}
 
-		if propsObj != nil {
-			if propsObj.Version != "" && propsObj.ArtifactID != "" {
-				// TODO: if there is no parentPkg (no java manifest) one of these poms could be the parent. We should discover the right parent and attach the correct info accordingly to each discovered package
-
-				// keep the artifact name within the virtual path if this package does not match the parent package
-				vPathSuffix := ""
-				if !strings.HasPrefix(propsObj.ArtifactID, parentPkg.Name) {
-					vPathSuffix += ":" + propsObj.ArtifactID
-				}
-				virtualPath := j.virtualPath + vPathSuffix
-
-				// discovered props = new package
-				p := pkg.Package{
-					Name:         propsObj.ArtifactID,
-					Version:      propsObj.Version,
-					Language:     pkg.Java,
-					Type:         pkg.JavaPkg,
-					MetadataType: pkg.JavaMetadataType,
-					Metadata: pkg.JavaMetadata{
-						VirtualPath:   virtualPath,
-						PomProperties: propsObj,
-						Parent:        parentPkg,
-					},
-				}
+		if propsObj == nil {
+			continue
+		}
 
-				pkgKey := uniquePkgKey(&p)
+		if propsObj.Version != "" && propsObj.ArtifactID != "" {
```

Comment:
is it possible to continue correctly if there is a `Version` but there is no `ArtifactID` ?

Reply:
@alfredodeza Fair question, tbh I'm not sure. This set of methods needs **_a lot_** of work: to simplify, to more clearly express business concerns, to reduce possible states, etc. I have a branch set to the side where I've started this, but wanted to keep the scope down for this impending release.